admin/cluster_add_nodes.py

```diff
@@ -0,0 +1,203 @@
+
+from twisted.internet.defer import DeferredList, inlineCallbacks
+from twisted.python.usage import UsageError
+from twisted.python.filepath import FilePath
+
+from flocker.provision._ca import Certificates
+from flocker.provision._common import Cluster
+
+from .acceptance import (
+    capture_journal,
+    capture_upstart,
+    eliot_output,
+    get_default_volume_size,
+    make_managed_nodes,
+    save_backend_configuration,
+)
+from .cluster_setup import (
+    RunOptions as SetupOptions,
+    make_client,
+    save_environment,
+    save_managed_config,
+    wait_for_nodes,
+)
+
+
+class RunOptions(SetupOptions):
+    optParameters = [
+        ['purpose', None, 'testing',
+         "Purpose of the cluster recorded in its metadata where possible."],
+        ['tag', None, None,
+         "Tag used in names of the existing cluster nodes."],
+        ['control-node', None, None,
+         "The address of the cluster's control node."],
+        ['cert-directory', None, None,
+         "Directory with the cluster certificates. "],
+        ['number-of-nodes', None, 1,
+         "Number of new nodes to create.", int],
+        ['starting-index', None, None,
+         "Starting index to use in names of new nodes.", int],
+    ]
+
+    synopsis = 'Usage: add-cluster-nodes [options]'
+
+    def __init__(self, top_level):
+        """
+        :param FilePath top_level: The top-level of the Flocker repository.
+        :param reactor: The reactor.
+        :param Deferred ready: A deferred to fire...
+        """
+        super(RunOptions, self).__init__(top_level)
+        self._remove_options(['no-keep'])
+
+    def _remove_options(self, to_remove):
+        """
+        Remove the given options that are defined in the parent classes.
+
+        :param to_remove: The options to rmeove.
+        :type to_remove: list of str
+
+        .. note::
+            Option names should be given as is,
+            parameter names should have '=' suffix.
+        """
+        self.longOpt = [opt for opt in self.longOpt if opt not in to_remove]
+
+    def postOptions(self):
+        if not self['control-node']:
+            raise UsageError("Control node address must be provided.")
+        if self.get('cert-directory') is None:
+            raise UsageError("Certificate directory must be set.")
+        if self.get('tag') is None:
+            raise UsageError("Tag must be specified.")
+
+        # This is run last as it creates the actual "runner" object
+        # based on the provided parameters.
+        super(RunOptions, self).postOptions()
+
+    def _check_cert_directory(self):
+        cert_path = FilePath(self['cert-directory'])
+        self['cert-directory'] = cert_path
+        if not cert_path.exists():
+            raise UsageError("{} does not exist".format(cert_path.path))
+        if not cert_path.isdir():
+            raise UsageError("{} is not a directory".format(cert_path.path))
+
+
+@inlineCallbacks
```

Comment:
I do see a yield in here, but the comment about inlineCallbacks being 'banned' applies still